ansible/terraform/shakenfist/setup.tf

```diff
@@ -0,0 +1,135 @@
+variable "namespace" {
+  description = "The Shaken Fist namespace to create"
+}
+
+variable "system_key" {
+  description = "The key for the Shaken Fist system namespace"
+}
+
+variable "ssh_key" {
+  description = "SSH public key text for ansible instance authentication"
+}
+
+variable "uniqifier" {
+  description = "A unique string to prefix hostnames with"
+}
+
+// Create new namespace using Shaken Fist "system" privilege.
+provider "shakenfist" {
+  alias      = "system"
+  server_url = "http://localhost:13000"
+  namespace  = "system"
+  key        = var.system_key
+}
+
+resource "shakenfist_namespace" "ci" {
+  provider = shakenfist.system
+  name     = var.namespace
+}
+
+resource "shakenfist_key" "cikey" {
+  provider  = shakenfist.system
+  namespace = shakenfist_namespace.ci.name
+  keyname   = "robot"
+  key       = "aMuop7IS6eithu9Ohqu4"
```

Comment:
Should we randomise this? Print it in a message at the end of the ansible script?

Reply:
Its never used outside this terraform, so I don't think we even need to output it? How do I generate a random string in terraform?

Reply:
For a publicly exposed cloud it could be an issue. Very simple to get a random string.
Answer: [random string](https://lmgtfy.com/?q=terraform+generate+random+string)

Reply:
Fixed.